front/app/modules/commercial/granular_permissions/admin/components/UserFieldSelection/UserFieldSelection.tsx

```diff
@@ -0,0 +1,160 @@
+
+// api
+import useAddPermissionCustomField from 'api/permissions_custom_fields/useAddPermissionsCustomField';
+import usePermissionsCustomFields from 'api/permissions_custom_fields/usePermissionsCustomFields';
+import useDeletePermissionsCustomField from 'api/permissions_custom_fields/useDeletePermissionsCustomField';
+import useUpdatePermissionsCustomField from 'api/permissions_custom_fields/useUpdatePermissionsCustomField';
+
+// utils
+import { FormattedMessage } from 'utils/cl-intl';
+import FormattedMessageComponent from 'utils/cl-intl/FormattedMessage';
+
+// services
+import { IPermissionData } from 'services/actionPermissions';
+import useUserCustomFields from 'hooks/useUserCustomFields';
+import useLocale from 'hooks/useLocale';
+import { isNilOrError } from 'utils/helperUtils';
+import { IUserCustomFieldData } from 'services/userCustomFields';
+
+type UserFieldSelectionProps = {
+  permission: IPermissionData;
+  projectId?: string | null;
+  phaseId?: string | null;
+  initiativeContext?: boolean;
+};
+
+const UserFieldSelection = ({
+  permission,
+  projectId,
+  phaseId,
+  initiativeContext,
+}: UserFieldSelectionProps) => {
+  const globalRegistrationFields = useUserCustomFields();
+  const initialFields = usePermissionsCustomFields({
+    projectId,
+    phaseId,
+    initiativeContext,
+    action: permission.attributes.action,
+  });
+  const { mutate: addPermissionCustomField } = useAddPermissionCustomField();
+  const { mutate: updatePermissionCustomField } =
+    useUpdatePermissionsCustomField();
+  const { mutate: deletePermissionsCustomField } =
+    useDeletePermissionsCustomField();
+  const locale = useLocale();
+  const [showSelectionModal, setShowSelectionModal] = useState(false);
+  const initialFieldArray = initialFields?.data?.data;
+
+  const handleAddField = (field: IUserCustomFieldData) => {
+    addPermissionCustomField({
+      custom_field_id: field.id,
+      required: false,
+      phaseId,
+      initiativeContext,
+      projectId,
+      action: permission.attributes.action,
+    });
+  };
+
+  const handleDeleteField = (fieldId: string) => {
+    deletePermissionsCustomField(fieldId);
+  };
+
+  if (isNilOrError(locale)) {
+    return null;
+  }
+
+  return (
+    <Box>
+      <Title variant="h4" color="primary" style={{ fontWeight: 600 }}>
+        <FormattedMessage {...messages.userQuestionTitle} />
+      </Title>
+      <Text fontSize="s" color="coolGrey600" pb="8px">
+        <FormattedMessage {...messages.userFieldsSelectionDescription} />
+      </Text>
+      <Box>
+        {initialFieldArray?.map((field) => (
+          <Box
+            display="flex"
+            justifyContent="space-between"
+            key={field.id}
+            py="0px"
+            borderTop="solid"
+            borderWidth="1px"
+            borderColor={colors.grey300}
+          >
+            <Text color="primary">
+              {
+                globalRegistrationFields?.find(
```

Comment:
This whole bit could maybe also be moved into a variable or function? Kind of hard to read